### PR TITLE
Create BJEDIS-ABNT-Number

### PR DESCRIPTION
**Pull request by ftir-mc**
﻿<?xml version="1.0" encoding="utf-8"?>
<style xmlns="http://purl.org/net/xbiblio/csl" class="in-text" version="1.0" demote-non-dropping-particle="never" default-locale="en-US">
  
  <info>
    <title>BJEDIS-ABNT (numeric, English)</title>
    <id>http://www.zotero.org/styles/bjedis-abnt</id>
    <link href="http://www.zotero.org/styles/bjedis-abnt" rel="self"/>
    <author>
      <name>Fernando Gomes</name>
      <email>fgsj@ufrj.br</email>
    </author>
    <category citation-format="numeric"/>
    <category field="generic-base"/>
    <summary>Style based on ABNT numeric, V1.1</summary>
    <updated>2021-02-09T15:04:30+00:00</updated>
    <rights license="http://creativecommons.org/licenses/by-sa/3.0/">This work is licensed under a Creative Commons Attribution-ShareAlike 3.0 License</rights>
  </info>
  <locale>
    <terms>
      <term name="no date">[no date]</term>
      <term name="in">in</term>
      <term name="online">online</term>
      <term name="accessed">accessed</term>
      <term name="retrieved">Available</term>
      <term name="from">from</term>
    </terms>
  </locale>
  <macro name="author">
    <names variable="author">
      <name and="text" delimiter-precedes-last="never" et-al-min="19" et-al-use-first="19" name-as-sort-order="all">
        <name-part name="family" text-case="uppercase"/>
        <name-part name="given"/>
      </name>
    </names>
  </macro>
  <macro name="editor">
    <names variable="editor">
      <name and="text" name-as-sort-order="all" sort-separator=", " delimiter=", " delimiter-precedes-last="never">
        <name-part name="family" text-case="uppercase"/>
        <name-part name="given"/>
      </name>
      <label prefix=" (" form="short" suffix=".)"/>
    </names>
  </macro>
  <macro name="translator">
    <names variable="translator">
      <name and="text" name-as-sort-order="all" sort-separator=", " delimiter=", " delimiter-precedes-last="never">
        <name-part name="family" text-case="uppercase"/>
        <name-part name="given"/>
      </name>
      <label prefix=" (" form="short" suffix=".)"/>
    </names>
  </macro>
  <macro name="responsability">
    <choose>
      <if variable="author editor translator" match="any">
        <choose>
          <if variable="author">
            <text macro="author"/>
          </if>
          <else-if variable="editor">
            <text macro="editor"/>
          </else-if>
          <else>
            <text macro="translator"/>
          </else>
        </choose>
      </if>
    </choose>
  </macro>
  <macro name="container-author">
    <names variable="container-author">
      <name and="text" name-as-sort-order="all" sort-separator=", " delimiter=", " delimiter-precedes-last="never">
        <name-part name="family" text-case="uppercase"/>
        <name-part name="given"/>
      </name>
    </names>
  </macro>
  <macro name="container-responsability">
    <choose>
      <if variable="container-author editor translator" match="any">
        <choose>
          <if variable="container-author">
            <text macro="container-author"/>
          </if>
          <else-if variable="editor">
            <text macro="editor"/>
          </else-if>
          <else>
            <text macro="translator"/>
          </else>
        </choose>
      </if>
    </choose>
  </macro>
  <macro name="year-date">
    <choose>
      <if variable="issued">
        <date variable="issued">
          <date-part name="year" form="long"/>
        </date>
      </if>
      <else>
        <text term="no date"/>
      </else>
    </choose>
  </macro>
  <macro name="title">
    <choose>
      <if type="book thesis map motion_picture song manuscript" match="any">
        <text variable="title" font-style="normal" font-weight="bold"/>
      </if>
      <else-if type="paper-conference speech chapter article-journal article-magazine article-newspaper entry entry-dictionary entry-encyclopedia post-weblog post webpage broadcast" match="any">
        <text variable="title" font-weight="normal" suffix=". "/>
        <choose>
          <if type="chapter paper-conference" match="any">
            <text term="in" text-case="capitalize-first" suffix=" : "/>
          </if>
        </choose>
        <choose>
          <if variable="container-author editor translator" match="any">
            <text macro="container-responsability"/>
            <choose>
              <if variable="container-title event" match="any">
                <text value=", "/>
              </if>
            </choose>
          </if>
        </choose>
        <choose>
          <if variable="container-title">
            <text variable="container-title" font-style="normal" font-weight="bold"/>
          </if>
          <else>
            <text variable="event" font-style="italic"/>
          </else>
        </choose>
      </else-if>
      <else-if type="report">
        <text variable="number" suffix=": "/>
        <text variable="title" font-style="italic"/>
      </else-if>
      <else-if type="patent">
        <text variable="title"/>
      </else-if>
      <else>
        <text variable="title" font-style="italic"/>
      </else>
    </choose>
    <choose>
      <if variable="URL">
        <text term="online" prefix=" [" suffix="]"/>
      </if>
    </choose>
  </macro>
  <macro name="number">
    <text variable="number"/>
  </macro>
  <macro name="medium">
    <text variable="medium" prefix=" [" suffix="]"/>
  </macro>
  <macro name="version">
    <text variable="version"/>
  </macro>
  <macro name="genre">
    <choose>
      <if type="map">
        <choose>
          <if variable="genre">
            <text variable="genre" prefix="[" suffix="]"/>
          </if>
          <else>
            <text value="map" prefix="[" suffix="]"/>
          </else>
        </choose>
      </if>
      <else>
        <text variable="genre"/>
      </else>
    </choose>
  </macro>
  <macro name="date">
    <choose>
      <if variable="issued">
        <date date-parts="year" form="numeric" variable="issued">
          
          
          <date-part name="year"/>
        </date>
      </if>
    </choose>
  </macro>
  <macro name="edition">
    <text variable="edition" form="long"/>
  </macro>
  <macro name="publisher-group">
    <group delimiter=" : ">
      <text variable="publisher"/>
      <text variable="publisher-place"/>
    </group>
  </macro>
  <macro name="issue">
    <group delimiter=", ">
      <text variable="volume" prefix="v. "/>
      <choose>
        <if variable="volume">
          <text variable="issue" prefix="n. "/>
          <text variable="page" prefix="p. "/>
        </if>
        <else-if variable="issue">
          <text variable="issue" prefix="No. "/>
          <text variable="page" prefix="p. "/>
        </else-if>
        <else>
          <text variable="page" prefix="P. "/>
        </else>
      </choose>
    </group>
  </macro>
  <macro name="accessed">
    <choose>
      <if variable="URL">
        <group>
          <text term="accessed" text-case="capitalize-first"/>
          <date variable="accessed">
            <date-part name="day" prefix=" "/>
            <date-part name="month" prefix=" "/>
            <date-part name="year" prefix=" "/>
          </date>
        </group>
      </if>
    </choose>
  </macro>
  <macro name="collection">
    <group delimiter=", ">
      <text variable="collection-title"/>
      <text variable="collection-number"/>
    </group>
  </macro>
  <macro name="page">
    <choose>
      <if type="book thesis manuscript" match="any">
        <text variable="number-of-pages" suffix=" p"/>
      </if>
      <else-if type="chapter paper-conference article-newspaper" match="any">
        <text variable="page" prefix="p. "/>
      </else-if>
      <else-if type="report patent" match="any">
        <text variable="page" suffix=" p"/>
      </else-if>
    </choose>
  </macro>
  <macro name="isbn">
    <text variable="ISBN" prefix="ISBN "/>
  </macro>
  <macro name="doi">
    <text variable="DOI" prefix="DOI "/>
  </macro>
  <macro name="url">
    <choose>
      <if variable="URL">
        <group>
          <text term="retrieved" suffix=" " text-case="capitalize-first"/>
          <text term="from" suffix=": "/>
          <text variable="URL" suffix=". "/>
        </group>
      </if>
    </choose>
  </macro>
  <macro name="archive">
    <group delimiter=": ">
      <text variable="archive"/>
      <text macro="archive_location"/>
    </group>
  </macro>
  <macro name="archive_location">
    <choose>
      <if variable="archive_location">
        <text variable="archive_location"/>
      </if>
      <else>
        <text variable="call-number"/>
      </else>
    </choose>
  </macro>
  <macro name="abstract">
    <text variable="abstract" form="short"/>
  </macro>
  <macro name="note">
    <text variable="note"/>
  </macro>
  <citation collapse="citation-number" after-collapse-delimiter="; ">
    <sort>
      <key variable="citation-number"/>
    </sort>
    <layout prefix="(" suffix=")" delimiter=", ">
      <group delimiter=", ">
        <text variable="citation-number"/>
        <group>
          <label variable="locator" suffix=". " form="short" strip-periods="true"/>
          <text variable="locator"/>
        </group>
      </group>
    </layout>
  </citation>
  <bibliography>
    <sort>
      <key variable="citation-number"/>
    </sort>
    <layout>
      <text variable="citation-number" display="left-margin" suffix=". "/>
      <choose>
        <if type="book map" match="any">
          <group display="right-inline">
            <text macro="responsability" suffix=". "/>
            <text macro="title" suffix=". "/>
            <text macro="genre" suffix=". "/>
            <text macro="edition" suffix=". "/>
            <text macro="publisher-group" suffix=", "/>
            <text macro="year-date" suffix=". "/>
            <text macro="accessed" suffix=". "/>
            <text macro="collection" suffix=". "/>
            <text macro="isbn" suffix=". "/>
            <text macro="url"/>
          </group>
        </if>
        <else-if type="article-journal article-magazine" match="any">
          <group display="right-inline">
            <text macro="responsability" suffix=". "/>
            <text macro="title" font-weight="bold" suffix=". "/>
            <text macro="edition" suffix=". "/>
            <text macro="issue" suffix=". "/>
            <text macro="date" suffix=". "/>
            <text macro="doi" suffix=". "/>
            <text macro="url"/>
            <text macro="accessed" suffix=". "/>
          </group>
        </else-if>
        <else-if type="article-newspaper">
          <group display="right-inline">
            <text macro="responsability" suffix=". "/>
            <text macro="title" suffix=". "/>
            <text macro="edition" suffix=". "/>
            <text macro="publisher-group" suffix=", "/>
            <text macro="date" suffix=". "/>
            <text macro="page" suffix=". "/>
            <text macro="accessed" suffix=". "/>
            <text macro="url"/>
          </group>
        </else-if>
        <else-if type="chapter entry entry-dictionary entry-encyclopedia" match="any">
          <group display="right-inline">
            <text macro="responsability" suffix=". "/>
            <text macro="title" font-style="normal" suffix=". "/>
            <text macro="edition" suffix=". "/>
            <text macro="publisher-group" suffix=", "/>
            <text macro="year-date" suffix=". "/>
            <text macro="page" suffix=". "/>
            <text macro="collection" suffix=". "/>
            <text macro="accessed" suffix=". "/>
            <text macro="isbn" suffix=". "/>
            <text macro="url"/>
          </group>
        </else-if>
        <else-if type="speech">
          <group display="right-inline">
            <text macro="responsability" suffix=". "/>
            <text macro="title" suffix=". "/>
            <text macro="genre" suffix=". "/>
            <text macro="publisher-group" suffix=". "/>
            <text macro="date" suffix=". "/>
            <text macro="accessed" suffix=". "/>
            <text macro="page" suffix=". "/>
            <text macro="url"/>
          </group>
        </else-if>
        <else-if type="paper-conference">
          <group display="right-inline">
            <text macro="responsability" suffix=". "/>
            <text macro="title" suffix=". "/>
            <text macro="genre" suffix=". "/>
            <text macro="publisher-group" suffix=", "/>
            <text macro="date" suffix=". "/>
            <text macro="page" suffix=". "/>
            <text macro="accessed" suffix=". "/>
            <text macro="collection" suffix=". "/>
            <text macro="isbn" suffix=". "/>
            <text macro="url"/>
          </group>
        </else-if>
        <else-if type="thesis">
          <group display="right-inline">
            <text macro="responsability" suffix=". "/>
            <text macro="title" suffix=". "/>
            <text macro="genre" suffix=". "/>
            <text macro="publisher-group" suffix=", "/>
            <text macro="year-date" suffix=". "/>
            <text macro="accessed" suffix=". "/>
            <text macro="url"/>
          </group>
        </else-if>
        <else-if type="post-weblog post webpage" match="any">
          <group display="right-inline">
            <text macro="responsability" suffix=". "/>
            <text macro="title" suffix=". "/>
            <text macro="date" suffix=". "/>
            <text macro="accessed" suffix=". "/>
            <text macro="url"/>
          </group>
        </else-if>
        <else-if type="broadcast motion_picture song" match="any">
          <group display="right-inline">
            <text macro="responsability" suffix=". "/>
            <text macro="title" suffix=". "/>
            <text macro="medium" suffix=". "/>
            <text macro="publisher-group" suffix=", "/>
            <text macro="date" suffix=". "/>
            <text macro="accessed" suffix=". "/>
            <text macro="collection" suffix=". "/>
            <text macro="isbn" suffix=". "/>
            <text macro="url"/>
          </group>
        </else-if>
        <else-if type="report" match="any">
          <group display="right-inline">
            <text macro="responsability" suffix=". "/>
            <text macro="title" suffix=". "/>
            <text macro="genre" suffix=". "/>
            <text macro="edition" suffix=". "/>
            <text macro="publisher-group" suffix=", "/>
            <text macro="year-date" suffix=". "/>
            <text macro="accessed" suffix=". "/>
            <text macro="collection" suffix=". "/>
            <text macro="url"/>
          </group>
        </else-if>
        <else-if type="manuscript" match="any">
          <group display="right-inline">
            <text macro="responsability" suffix=". "/>
            <text macro="title" suffix=". "/>
            <text macro="genre" suffix=". "/>
            <text macro="edition" suffix=". "/>
            <text macro="publisher-group" suffix=", "/>
            <text macro="year-date" suffix=". "/>
            <text macro="accessed" suffix=". "/>
            <text macro="collection" suffix=". "/>
            <text macro="url"/>
          </group>
        </else-if>
        <else-if type="patent">
          <group display="right-inline">
            <text macro="responsability" suffix=". "/>
            <text macro="title" suffix=". "/>
            <text macro="number" suffix=". "/>
            <text macro="date" suffix=". "/>
            <text macro="publisher-group" suffix=". "/>
            <text macro="accessed" suffix=". "/>
            <text macro="collection" suffix=". "/>
            <text macro="url"/>
          </group>
        </else-if>
        <else>
          <group display="right-inline">
            <text macro="responsability" suffix=". "/>
            <text macro="title" suffix=". "/>
            <text macro="version" suffix=". "/>
            <text macro="medium" suffix=". "/>
            <text macro="genre" suffix=". "/>
            <text macro="date" suffix=". "/>
            <text macro="edition" suffix=". "/>
            <text macro="publisher-group" suffix=". "/>
            <text macro="number" suffix=". "/>
            <text macro="accessed" suffix=". "/>
            <text macro="collection" suffix=". "/>
            <text macro="page" suffix=". "/>
            <text macro="isbn" suffix=". "/>
            <text macro="url"/>
          </group>
        </else>
      </choose>
      <group display="right-inline">
        <text macro="archive"/>
      </group>
      
      <group display="right-inline">
        <text macro="note"/>
      </group>
    </layout>
  </bibliography>
</style>

### CODE REVIEW

**Comment by rmzelle** [comment]
@ftir-mc, thanks. Is this style for the Brazilian Journal of Experimental Design, Data Analysis and Inferential Statistics  (BJEDIS) at https://bjedis.org/?

**Comment by ftir-mc** [comment]
Dear Dr. Zelle,
Yes, it is!
Thank you for answering my message!
Best regards,
Fernando



Em sex., 12 de fev. de 2021 às 02:06, Rintze M. Zelle <
notifications@github.com> escreveu:

**Comment by rmzelle** [comment]
@ftir-mc, thanks. And do you know which CSL style you based this one on? Was it https://github.com/citation-style-language/styles/blob/master/associacao-brasileira-de-normas-tecnicas.csl (https://www.zotero.org/styles?q=id%3Aassociacao-brasileira-de-normas-tecnicas)?

**Comment by ftir-mc** [comment]
Yes, it is.
Thanks
Fernando

Em dom, 14 de fev de 2021 01:26, Rintze M. Zelle <notifications@github.com>
escreveu:

**Comment by ftir-mc** [comment]
Thank you very much!
B R
Fernando

Em seg, 15 de fev de 2021 00:58, github-actions[bot] <
notifications@github.com> escreveu:

File: brazilian-journal-of-experimental-design-data-analysis-and-inferential-statistics.csl [hunk 1]
```diff
@@ -0,0 +1,492 @@
...
+        </group>
+      </group>
+    </layout>
+  </citation>
+  <bibliography>
+    <sort>
+      <key variable="citation-number"/>
+    </sort>
+    <layout>
+      <text variable="citation-number" display="left-margin" suffix=". "/>
```
Comment (adam3smith) [inline brazilian-journal-of-experimental-design-data-analysis-and-inferential-statistics.csl:313]: Is there are reason you're using the right/left display element? It looks like you're just trying to produce `second-field-align="flush"`? in a much more complex way.

File: brazilian-journal-of-experimental-design-data-analysis-and-inferential-statistics.csl [hunk 2]
```diff
@@ -0,0 +1,492 @@
...
+      <term name="in">in</term>
+      <term name="online">online</term>
+      <term name="accessed">accessed</term>
+      <term name="retrieved">Available</term>
+      <term name="from">from</term>
+    </terms>
+  </locale>
+  <macro name="author">
+    <names variable="author">
+      <name and="text" delimiter-precedes-last="never" et-al-min="19" et-al-use-first="19" name-as-sort-order="all">
```
Comment (adam3smith) [inline brazilian-journal-of-experimental-design-data-analysis-and-inferential-statistics.csl:32]: `et-al-use-first` should always be at least one lower the `et-al-min` -- it doesn't make sense to have et al with 19 authors but then also print the first 19 authors followed by et al.
We generally prefer to have et al in the citation or bibliography section where possible.

File: brazilian-journal-of-experimental-design-data-analysis-and-inferential-statistics.csl [hunk 3]
```diff
@@ -0,0 +1,492 @@
...
+        <name-part name="given"/>
+      </name>
+      <label prefix=" (" form="short" suffix=".)"/>
+    </names>
+  </macro>
+  <macro name="translator">
+    <names variable="translator">
+      <name and="text" name-as-sort-order="all" sort-separator=", " delimiter=", " delimiter-precedes-last="never">
+        <name-part name="family" text-case="uppercase"/>
+        <name-part name="given"/>
```
Comment (adam3smith) [inline brazilian-journal-of-experimental-design-data-analysis-and-inferential-statistics.csl:51]: all these given name name-parts don't do anything (you only need to use `name-part` when you apply additional casing or formatting). Please removes

File: brazilian-journal-of-experimental-design-data-analysis-and-inferential-statistics.csl [hunk 4]
```diff
@@ -0,0 +1,492 @@
...
+      <else>
+        <text variable="call-number"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="abstract">
+    <text variable="abstract" form="short"/>
+  </macro>
+  <macro name="note">
+    <text variable="note"/>
```
Comment (adam3smith) [inline brazilian-journal-of-experimental-design-data-analysis-and-inferential-statistics.csl:292]: We much prefer not to use `note` in citation styles, as it's not well defined and used very differently across reference managers. What are you trying to use this for?

**Review by adam3smith — COMMENTED** [review]
Thanks -- a couple of relatively small change requests and some questions about the style

**Comment by ftir-mc** [comment]
Dear Sebastian,
I fully agree with all of your pointed issues.
I was simply using the CLS Editor based on the ASTM sample.
Actually, I don't know how to fix these issues.
Could you guide me on how can I solve them?
Thanks in advance,

Fernando



Em sáb., 20 de fev. de 2021 às 01:25, Sebastian Karcher <
notifications@github.com> escreveu:

**Comment by ftir-mc** [comment]
Dear Sebastian and Github Team,
I really would like to count on you guiding me on how to solve the issues
you pointed out in the last email.
Could you help me?
Thanks in advance,
Fernando



Em seg., 22 de fev. de 2021 às 08:42, Fernando Gomes Souza Jr <
fgsj74@gmail.com> escreveu:

**Comment by adam3smith** [comment]
There are some questions I asked, so answering those would help, but otherwise I gave pretty specific guidance on what to change. If you have questions about individual points, I'm happy to help, but I'm afraid fixing this myself line-by-line is beyond what we can provide as support.

File: brazilian-journal-of-experimental-design-data-analysis-and-inferential-statistics.csl (hunk 1 above)
Comment (ftir-mc) [reply brazilian-journal-of-experimental-design-data-analysis-and-inferential-statistics.csl:313]: There is not. I've just used this option from the editor page.

File: brazilian-journal-of-experimental-design-data-analysis-and-inferential-statistics.csl (hunk 2 above)
Comment (ftir-mc) [reply brazilian-journal-of-experimental-design-data-analysis-and-inferential-statistics.csl:32]: Thank you so much. We've corrected it.
<name and="text" delimiter-precedes-last="never" et-al-min="19" et-al-use-first="18" name-as-sort-order="all">
Our publications can receive up to 19 authors. So we would like to keep this number.

File: brazilian-journal-of-experimental-design-data-analysis-and-inferential-statistics.csl (hunk 3 above)
Comment (ftir-mc) [reply brazilian-journal-of-experimental-design-data-analysis-and-inferential-statistics.csl:51]: Thank you! They were removed.

**Comment by ftir-mc** [comment]
> u have

Thank you, dear Dr. Sebastian!
I did my best to address all of your observations.
BR
Fernando

**Review by ftir-mc — COMMENTED** [review]
Thank you all for kind support!

**Comment by ftir-mc** [comment]
Someone could help me?
I have no idea how to solve these issues.
Thanks in advance
Fernando



Em sex., 12 de mar. de 2021 às 07:41, github-actions[bot] <
***@***.***> escreveu:

**Comment by adam3smith** [comment]
@ftir-mc all looks good from our end now. If you could check that I didn't introduce any errors, we can go ahead and publish

**Comment by ftir-mc** [comment]
Dear Sebastian,
What great news! Thank you!
I've tried from my side using Zotero, and it worked fine!
Here you are the PDF file with references following the new BJEDIS style!
Thank you all again,
Fernando



Em sex., 12 de mar. de 2021 às 11:49, Sebastian Karcher <
***@***.***> escreveu:

**Comment by adam3smith** [comment]
Great, thanks!

**Comment by ftir-mc** [comment]
Dear Sebastian,
BJEDIS style could be published?
Is there anything else I need to do?
BR
Fernando

**Comment by adam3smith** [comment]
Styles go up typically within a couple of hours of us merging them. Yours is here: https://www.zotero.org/styles?q=id%3Abrazilian-journal-of-experimental-design-data-analysis-and-inferential-statistics

**Comment by ftir-mc** [comment]
Cool, thanks!
BR
Fernando

Em ter, 16 de mar de 2021 21:51, Sebastian Karcher ***@***.***>
escreveu:

**Comment by ftir-mc** [comment]
Dear Sebastian and GitHub Team,
I've just verified that the BJEDIS references do not appear sequentially.
I proved that the count was not sequential by comparing the same document
formatted for BJEDIS and Bentham.
Can you guide me on how to solve this counting problem?
Thanks,
Fernando

PS.: Here you are the format tests:
[image: Screenshot from 2021-03-23 09-20-32.png]




Em ter., 16 de mar. de 2021 às 21:51, Sebastian Karcher <
***@***.***> escreveu:

**Comment by adam3smith** [comment]
We can't see the screenshot unless you come here on github and paste it in (email doesn't work), but the style is definitely set to display references in the order in which they appear.

**Comment by adam3smith** [comment]
That would look like it's something on your end. Maybe you just need to hit refresh in the word processor, but looks correctly for me with BJEDIS
image

**Comment by ftir-mc** [comment]
Thank God!
I'll check it out!
Thank you, Sebastian!
BR
Fernando



Em ter., 23 de mar. de 2021 às 10:51, Sebastian Karcher <
***@***.***> escreveu:

**Pull request merged**